netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1241,6 +1276,13 @@ _net_ebpf_ext_process_redirect_verdict(
             }
             commit_layer_data = TRUE;
 
+            if (_net_ebpf_ext_sock_addr_is_connection_locally_redirected_by_others(connect_request, filter->filterId)) {
```

Comment:
Should we move this check even before we call `FwpsAcquireWritableLayerDataPointer`, actually before calling `_net_ebpf_ext_compare_destination_address()` ? If we are not redirecting, we dont need to call these functions?

Reply:
The `FWPS_CONNECT_REQUEST` struct is obtained from `FwpsAcquireWritableLayerDataPointer` which currently is called only if the eBPF program modifies `sock_addr`.  Assuming that most of the times the eBPF program will not redirect a connection, calling `FwpsAcquireWritableLayerDataPointer` unconditionally will incur CPU overhead. Thoughts?